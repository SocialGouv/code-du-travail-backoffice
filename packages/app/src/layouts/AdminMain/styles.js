import styled from "@emotion/styled";

export const Content = styled.div`
  display: flex;
  flex-grow: 1;
  overflow-y: ${p => (p.noScroll ? "hidden" : "scroll")};
`;
