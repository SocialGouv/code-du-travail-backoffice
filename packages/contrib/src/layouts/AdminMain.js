import React from "react";
import styled from "styled-components";

import AdminMenu from "./AdminMenu";
import Main from "./Main";

const Content = styled.div`
  flex-grow: 1;
  overflow-y: scroll;
`;

export default ({ children, hasBareContent, isLoading }) => (
  <Main isAdmin isHorizontal isLoading={isLoading}>
    <AdminMenu />
    {Boolean(hasBareContent) ? children : <Content>{children}</Content>}
  </Main>
);
