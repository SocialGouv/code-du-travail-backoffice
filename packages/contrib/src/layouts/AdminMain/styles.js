import styled from "@emotion/styled";

export const Content = styled.div`
  flex-grow: 1;
  overflow-y: ${p => (p.isScrollable ? "scroll" : "hidden")};
`;
