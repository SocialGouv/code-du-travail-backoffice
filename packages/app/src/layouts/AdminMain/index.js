import PropTypes from "prop-types";
import React from "react";
import { Flex } from "rebass";

import LoadingSpinner from "../../elements/LoadingSpinner";
import AdminMenuWithRouter from "../AdminMenu";
import Main from "../Main";
import { Content } from "./styles";

const AdminMain = ({ children, hasBareContent = false, isLoading = false, noScroll = false }) => {
  if (isLoading || children === undefined) {
    return (
      <Main isHorizontal>
        <AdminMenuWithRouter />
        <Flex alignItems="center" flexGrow="1" justifyContent="center">
          <LoadingSpinner color="#666666" />
        </Flex>
      </Main>
    );
  }

  return (
    <Main isHorizontal>
      <AdminMenuWithRouter />
      {hasBareContent ? (
        children
      ) : (
        <Content data-testid="content" noScroll={noScroll}>
          {children}
        </Content>
      )}
    </Main>
  );
};

AdminMain.propTypes = {
  children: PropTypes.element,
  hasBareContent: PropTypes.bool,
  isLoading: PropTypes.bool,
  noScroll: PropTypes.bool,
};

export default AdminMain;
