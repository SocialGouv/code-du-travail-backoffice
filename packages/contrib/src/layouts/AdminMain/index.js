import React from "react";

import AdminMenu from "../AdminMenu";
import Main from "../Main";

import { Content } from "./styles";

export default ({ children, hasBareContent = false, isLoading, isScrollable = true }) => (
  <Main isHorizontal isLoading={isLoading}>
    <AdminMenu />
    {hasBareContent ? children : <Content isScrollable={isScrollable}>{children}</Content>}
  </Main>
);
