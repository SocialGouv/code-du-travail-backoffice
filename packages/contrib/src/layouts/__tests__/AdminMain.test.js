import React from "react";
import { render } from "react-testing-library";
import AdminMain from "../AdminMain";

describe.skip("[Contrib] layouts/<AdminMain />", () => {
  it("should render", () => {
    const { container } = render(<AdminMain />);

    expect(container).toMatchSnapshot();
  });
});
