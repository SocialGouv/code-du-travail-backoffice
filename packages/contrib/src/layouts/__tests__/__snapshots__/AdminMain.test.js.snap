// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Contrib] layouts/<AdminMain /> should match snapshot 1`] = `
<div>
  <div
    class="Main__Container-q4q6nx-0 kSNCLj css-1usr1wh"
    style="height: 100vh;"
  >
    <div
      class="Main__Body-q4q6nx-1 bSllml css-1gmfemw"
    >
      <div
        class="Header__Container-sc-7cr55z-0 eHuDxq css-s3axrf"
      >
        <div
          class="Header__Brand-sc-7cr55z-1 knuMJV css-zkfaav"
        >
          <img
            alt="Code du travail numérique"
            aria-label="Bouton de retour au tableau de bord"
            class="Header__Logo-sc-7cr55z-2 kAIypM"
            src="/static/images/marianne.svg"
          />
          <div
            class="css-l3rx45"
          >
            <span
              class="Header__Title-sc-7cr55z-3 jGdlvq"
            >
              Code du travail numérique
            </span>
            <span
              class="Header__Subtitle-sc-7cr55z-4 kxPxIS"
            >
              Outil de contribution
            </span>
          </div>
        </div>
      </div>
      <div
        class="Main__Content-q4q6nx-2 iWRxVq css-1kw0tju"
      >
        <div
          class="AdminMenu__Container-t0dk93-0 bMnGdR css-l3rx45"
        >
          <div
            class="AdminMenu__Link-t0dk93-2 LnAQE"
          >
            Tableau de bord
          </div>
          <div
            class="AdminMenu__Link-t0dk93-2 kospmv"
          >
            Conventions
          </div>
          <div
            class="AdminMenu__Link-t0dk93-2 kospmv"
          >
            Étiquettes
          </div>
          <div
            class="AdminMenu__Link-t0dk93-2 kospmv"
          >
            Questions
          </div>
          <div
            class="AdminMenu__Link-t0dk93-2 kospmv"
          >
            Réponses
          </div>
          <div
            class="AdminMenu__Link-t0dk93-2 kospmv"
          >
            Réponses génériques
          </div>
          <div
            class="AdminMenu__Link-t0dk93-2 kospmv"
          >
            Unités
          </div>
          <div
            class="AdminMenu__Link-t0dk93-2 kospmv"
          >
            Utilisateurs
          </div>
          <div
            class="AdminMenu__Link-t0dk93-2 kospmv"
          >
            Zones
          </div>
          <h2
            class="Subtitle-sc-17ig8qv-0 AdminMenu__Subtitle-t0dk93-1 vshUS"
          >
            Maintenance
          </h2>
          <div
            class="AdminMenu__Link-t0dk93-2 kospmv"
          >
            Migrations
          </div>
        </div>
        <div
          class="AdminMain__Content-lizrru-0 dcXFX"
        >
          Some Content Text
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[Contrib] layouts/<AdminMain /> should match snapshot diff with \`hasBareContent\` prop 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,6 +1,6 @@
- <div>
+ <DocumentFragment>
    <div
      class=\\"Main__Container-q4q6nx-0 kSNCLj css-1usr1wh\\"
      style=\\"height: 100vh;\\"
    >
      <div
@@ -94,14 +94,10 @@
              class=\\"AdminMenu__Link-t0dk93-2 kospmv\\"
            >
              Migrations
            </div>
          </div>
-         <div
-           class=\\"AdminMain__Content-lizrru-0 dcXFX\\"
-         >
          Some Content Text
        </div>
      </div>
    </div>
-   </div>
- </div>
+ </DocumentFragment>"
`;

exports[`[Contrib] layouts/<AdminMain /> should match snapshot diff with \`isLoading\` prop 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,6 +1,6 @@
- <div>
+ <DocumentFragment>
    <div
      class=\\"Main__Container-q4q6nx-0 kSNCLj css-1usr1wh\\"
      style=\\"height: 100vh;\\"
    >
      <div
@@ -33,75 +33,23 @@
              </span>
            </div>
          </div>
        </div>
        <div
-         class=\\"Main__Content-q4q6nx-2 iWRxVq css-1kw0tju\\"
+         class=\\"Main__Content-q4q6nx-2 iWRxVq css-1g1x2nw\\"
        >
          <div
-           class=\\"AdminMenu__Container-t0dk93-0 bMnGdR css-l3rx45\\"
+           class=\\"LoadingSpinner__Container-eu7nmw-0 dEMrTz\\"
          >
            <div
-             class=\\"AdminMenu__Link-t0dk93-2 LnAQE\\"
-           >
-             Tableau de bord
-           </div>
+             class=\\"LoadingSpinner__Spinner-eu7nmw-1 sHZX\\"
+             color=\\"#666666\\"
+           />
            <div
-             class=\\"AdminMenu__Link-t0dk93-2 kospmv\\"
-           >
-             Conventions
+             class=\\"LoadingSpinner__Spinner-eu7nmw-1 LoadingSpinner__SpinnerNext-eu7nmw-2 daPQQA\\"
+             color=\\"#666666\\"
+           />
          </div>
-           <div
-             class=\\"AdminMenu__Link-t0dk93-2 kospmv\\"
-           >
-             Étiquettes
        </div>
-           <div
-             class=\\"AdminMenu__Link-t0dk93-2 kospmv\\"
-           >
-             Questions
      </div>
-           <div
-             class=\\"AdminMenu__Link-t0dk93-2 kospmv\\"
-           >
-             Réponses
    </div>
-           <div
-             class=\\"AdminMenu__Link-t0dk93-2 kospmv\\"
-           >
-             Réponses génériques
-           </div>
-           <div
-             class=\\"AdminMenu__Link-t0dk93-2 kospmv\\"
-           >
-             Unités
-           </div>
-           <div
-             class=\\"AdminMenu__Link-t0dk93-2 kospmv\\"
-           >
-             Utilisateurs
-           </div>
-           <div
-             class=\\"AdminMenu__Link-t0dk93-2 kospmv\\"
-           >
-             Zones
-           </div>
-           <h2
-             class=\\"Subtitle-sc-17ig8qv-0 AdminMenu__Subtitle-t0dk93-1 vshUS\\"
-           >
-             Maintenance
-           </h2>
-           <div
-             class=\\"AdminMenu__Link-t0dk93-2 kospmv\\"
-           >
-             Migrations
-           </div>
-         </div>
-         <div
-           class=\\"AdminMain__Content-lizrru-0 dcXFX\\"
-         >
-           Some Content Text
-         </div>
-       </div>
-     </div>
-   </div>
- </div>
+ </DocumentFragment>"
`;
